go/oasis-node/cmd/config/migrate/migrate.go

```diff
@@ -50,6 +54,86 @@ func pruneEmptyMaps(m map[string]interface{}) {
 	}
 }
 
+// Translate known P2P addresses from the old format into the new.
+func translateKnownP2P(in []string) ([]string, uint) {
+	// Old format is the Tendermint P2P address, which we can't translate
+	// automatically because it is hashed.
+	// New format is the node's P2P public key.
+	// However, we know both the old and new addresses of Oasis public
+	// seed nodes for the testnet and mainnet, so we can at least translate
+	// those.
+
+	known := map[string]string{
+		// Mainnet seed node.
+		"E27F6B7A350B4CC2B48A6CBE94B0A02B0DCB0BF3@35.199.49.168:26656": "H6u9MtuoWRKn5DKSgarj/dzr2Z9BsjuRHgRAoXITOcU=@35.199.49.168:26656",
+		// Testnet seed node.
+		"53572F689E5BACDD3C6527E6594EC49C8F3093F6@34.86.165.6:26656": "HcDFrTp/MqRHtju5bCx6TIhIMd6X/0ZQ3lUG73q5898=@34.86.165.6:26656",
+	}
+
+	var numUnknown uint
+
+	out := make([]string, len(in))
+	for idx, p2pOld := range in {
+		var p2pNew string
+
+		if knownNew, isKnown := known[p2pOld]; isKnown {
+			// New address is known, use it.
+			p2pNew = knownNew
+		} else {
+			// New address is not known, make sure the user replaces the ID.
+			s := strings.Split(p2pOld, "@")
+			addr := s[len(s)-1]
+			p2pNew = fmt.Sprintf("INSERT_P2P_PUBKEY_HERE@%s", addr)
```

Comment:
Using vars we can end up with values that are not expected. But I guess this is still ok, as we don't crash.
Ideally, we would migrate raw configs (fix values) and templates (keep values, show warnings) differently, but that is probably too much.
```
INSERT_P2P_PUBKEY_HERE@{{ p2p_address }}
INSERT_P2P_PUBKEY_HERE@${P2P_ADDRESS}
```

Reply:
I mean, it's not a 100% hands-off automatic process -- you are expected to check the resulting migrated config and make any changes if required :P